### ISSUE TITLE
"Value must be 8-bit" error

### ISSUE THREAD

**Issue by Wrathchild70** [rednex/rgbds#162]
In building the pokemon-yellow sources using the rgdbs tools compiled under 64-bit ubuntu, I receive the following error:
rgblink: error: audio/engine_2.asm(192) : Value must be 8-bit

This line (the 'ld a') uses a value from a macro:

`Music_PokeFluteInBattle:: ; 219cf (8:59cf)
        ; begin playing the "caught mon" sound effect
        ld a, SFX_CAUGHT_MON
        call PlaySoundWaitForCurrent
`

The macro creates a value from the offset from the first table entry:

`music_const: MACRO
\1 EQUS "((\2 - SFX_Headers_1) / 3)"
ENDM

        ; AUDIO_1
        music_const MUSIC_PALLET_TOWN,         Music_PalletTown
`

The same code built under Cygwin64 does not produce this error.
For that I think I used the 0.2.4 downloaded binaries (in Oct last year).
Today I've tried the 0.25 64-bit binaries and they work ok too.

**Comment by AntonioND** [comment]
Are you building the current master branch? If so, I'd be surprised that disassembly could be built correctly.

**Comment by AntonioND** [comment]
Yeah, I've checked and the commit that broke it is the same one that broke the other ones, fa962b94702787480203331f44b036199eaf7218.

It's not the same problem, though. In the other case, all the code compiled correctly and the only problem is that the sections were being placed in a different order than expected.

**Comment by Wrathchild70** [comment]
I'd used "git clone git://github.com/bentley/rgbds.git" to obtain the sources yesterday.
Just tried the 0.2.5 sources from releases and that works fine under ubuntu

**Comment by AntonioND** [comment]
Anyway, I've located the problem. `SFX_Caught_Mon` is in the file `audio/headers/sfxheaders2.asm`, which is in section `SECTION "Sound Effect Headers 2", ROMX, BANK[AUDIO_2]` as defined in file `audio.asm`.

`music_const SFX_CAUGHT_MON, SFX_Caught_Mon` expands to the following expression: `((SFX_Caught_Mon - SFX_Headers_1) / 3)`.

However, `SFX_Headers_1` is defined in file `audio/headers/sfxheaders1.asm` and it is in section `SECTION "Sound Effect Headers 1", ROMX, BANK[AUDIO_1]` as defined in file `audio.asm`.

In short, this expression is trying to calculate the difference between two labels that are in floating sections (the bank is fixed, but the address isn't). Even though that error could be more descriptive, it is correct.

**Comment by AntonioND** [comment]
That should be easily fixable by adding a linkerscript like in pret/pokered.
